dspy/dsp/utils/settings.py

```diff
@@ -22,6 +22,7 @@
     disable_history=False,
     track_usage=False,
     usage_tracker=None,
+    adapter_retry_count=0, # number of times to retry the adapter call
```

Comment:
For thinking: 

- Shall we just call this `max_retry`? `adapter_retry_count` could be mysterious for DSPy beginners. 
- Shall we default to 1 instead of 0? It would be helpful to collect data on the success ratio of 1 retry, 2 retries when the first call fails. A small model like llama-3b might provide useful data for us.

Reply:
We have num_retries in LM. So I think it's confusing if we can use retry without any prefix. Maybe `max_parse_retries`?